src/module/address/infrastructure/address.repository.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from "@nestjs/common";
+import { Repository, DataSource } from "typeorm";
+import { Address } from "../domain/address.entity";
+import { IAddressRepository } from "../application/repository/address.repository.interface";
+import { ICreateAddress, IUpdateAddress } from "../interface/address.interface";
+
+@Injectable()
+export class AddressRepository implements IAddressRepository {
+  repository: Repository<Address>
+  constructor(private readonly dataSource: DataSource) {
+    this.repository = this.dataSource.getRepository(Address)
+  }
+
+  async findOne(id: number): Promise<Address> {
+    return await this.repository.findOne({
+      where: { id: id }
+    })
+  }
+
+  async create(data: ICreateAddress): Promise<Address> {
```

Comment:
You should pass an  Address without id. But it must be an Address instance. There is no need for de ICreateAddress interface